### ISSUE TITLE
BUG: CHGM/hyp1f1 broken for large negative inputs

### ISSUE THREAD

**Issue by Luthaf** [scipy/scipy#17120]
### Describe your issue.

The CHGM fortran subroutine used in hyp1f1 has a bug for large negative values. What happens is that if the series converges, the code correctly multiplies by `exp(x)` after computing hyp1f1(-x) (https://github.com/scipy/scipy/blob/ac815d40de0819497dfabd27046c3b605dae447f/scipy/special/specfun/specfun.f#L5398-L5402), but fails to do so if the serie does not converge.

This makes the code below print:
```
x = -350
scipy = -5.224090831922376e-20
mpmath = -5.22409083192238e-20

x = -355
scipy = -6.868550128785144e+134
mpmath = -4.59540715981337e-20
```

I have a fix for this, I am opening this issue to reference in the tests and PR.

### Reproducing Code Example

```python
import scipy.special
import mpmath

print("x = -350")
print("scipy =", scipy.special.hyp1f1(9, 8.5, -350))
print("mpmath =", mpmath.hyp1f1(9, 8.5, -350))

print("\nx = -355")
print("scipy =", scipy.special.hyp1f1(9, 8.5, -355))
print("mpmath =", mpmath.hyp1f1(9, 8.5, -355))
```

### Error message

```shell
/
```

### SciPy/NumPy/Python version information

1.9.1 1.23.3 sys.version_info(major=3, minor=10, micro=6, releaselevel='final', serial=0)

**Comment by j-bowhay** [comment]
Believe this is also tracked #14149 ?

**Comment by Luthaf** [comment]
Yes, this seems like the same issue. I'll check if my fix helps with the range of values in the other issue.

**Comment by Luthaf** [comment]
Unfortunately, #14149 seems like a different issue, where the terms of the series overflow a double. Printing `RG` gives:

```
RG = -23333.333333333332
RG = -26923076.92307693
RG = -50724637681.15943
RG = -88383838383838.39
RG = -1.3565891472868216e17
... a bunch more
RG = -2.6349541843694608e305
RG = -1.937284646184326e307
RG = -inf
```

My fix only applies when the series does converges, but not quite up to the hardcoded precision of 1e-15.
